api/utils/keys/yubikey.go

```diff
@@ -442,59 +473,78 @@ func parsePIVSlot(slotKey uint32) (piv.Slot, error) {
 	}
 }
 
+// selfSignedMetadataCertificate creates a self signed certificate to be stored in the
+// YubiKey's PIV slot. This certificate is purely used as metadata to determine when a
+// slot is in used by a Teleport Client and is not fit to be used in cryptographic operations.
+func selfSignedMetadataCertificate() (*x509.Certificate, error) {
+	// generate a small rsa key to quickly generate a metadata cert.
+	priv, err := rsa.GenerateKey(rand.Reader, 512)
```

Comment:
Does it need to be RSA? EC keys are already small and typically faster.

I know this likely doesn't matter, but a key this small makes me uneasy.